[dev] locating the system python2 to create the virtualenv

palazzem opened the pull request:
## What's this PR do?

Enforces the use of ``python2`` while creating the ``venv/`` virtualenv, during the execution of ``rake setup_env``.

## Motivation

Without the ``-p`` option, virtualenv uses the ``python`` system executable. Unfortunately I'm running a system (ArchLinux) that uses Python 3 as a default and because of that, the virtualenv is not properly created.

degemer commented:
Thanks @palazzem !
It seems that ArchLinux is the only distribution doing this (and caused PEP 394 https://www.python.org/dev/peps/pep-0394/). 😄 

I don't think the `which` is needed according to `virtualenv.py`:
```
    parser.add_option(
        '-p', '--python',
        dest='python',
        metavar='PYTHON_EXE',
        help='The Python interpreter to use, e.g., --python=python2.5 will use the python2.5 '
        'interpreter to create the new environment.  The default is the interpreter that '
        'virtualenv was installed with (%s)' % sys.executable)
```
Could you remove it and just use `-p python2` instead ? I tried on OS X (and it worked), but it would be perfect if you could double check on ArchLinux.

palazzem commented:
Hi @degemer , you're welcome!
I've updated and rebased the PR with your upstream. As you asked, I've double checked the ``Rakefile`` and it still works in ArchLinux.

Let me know if it is ready to merge,
Cheers! :D

degemer commented:
That's perfect, thanks again!
